test/components/tweets/form_test.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require "test_helper"
+require "view_component/test_case"
+
+module Tweets
+  class FormTest < ViewComponent::TestCase
+    include Motion::TestHelpers
+    let(:klass) { Tweets::Form }
+    let(:tweet) { Tweet.new }
+    let(:on_change) { callback_stub }
+    subject { klass.new(tweet: tweet, on_change: on_change) }
+
+    describe "rendering" do
+      it "shows form with textarea and submit button" do
+        render_inline(subject)
+        assert_selector "form" do
+          assert_selector "textarea"
+          assert_selector "input[type='submit']"
+        end
+      end
+
+      describe "when !disabled?" do
+        let(:tweet) { Tweet.new(content: "tweet") }
+        before do
+          refute subject.disabled?
```

Comment:
Note: you can't stub components and then render them.  Stubs add procs that can't be Marshal dumped.